pkg/errors/sdk_errors.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package errors
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+)
+
+var (
+	ErrJSONKeyNotFound = errors.New("response body expected error message json key not found")
+	ErrUnknown         = errors.New("unknown error")
+)
+
+const err = "error"
+
+// SDKError is an error type for Mainflux SDK.
+type SDKError interface {
+	Error
+	StatusCode() int
+}
+
+var _ SDKError = (*sdkError)(nil)
+
+type sdkError struct {
+	*customError
+	statusCode int
+}
+
+func (ce *sdkError) Error() string {
+	if ce == nil {
+		return ""
+	}
+	if ce.err == nil {
+		return ce.msg
+	}
+	return ce.msg + " : " + ce.err.Error()
+}
+
+func (ce *sdkError) Msg() string {
+	return ce.msg
+}
+
+func (ce *sdkError) Err() Error {
+	return ce.err
+}
+
```

Comment:
No need for these. Eventually, you can change `Error` to something like:
```go
func (ce *sdkError) Error() string {
	if ce == nil {
		return ""
	}
	if ce.customError == nil {
		return http.StatusText(ce.statusCode)
	}
	return fmt.Sprintf("Status: %s: %s", http.StatusText(ce.statusCode), ce.customError.Error())
}
```

Reply:
But in this, shouldn't we also return the `ce.msg`?

Reply:
`ce.msg` is returned anyways. Please take a look at the implementation of `ce.Error()`.